x-pack/plugins/security_solution/public/contract_get_started_page.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Observable } from 'rxjs';
+import { BehaviorSubject } from 'rxjs';
+import type { SecurityProductTypes } from './common/components/landing_page/get_started/configs';
+import type { StepId } from './common/components/landing_page/get_started/types';
+
+export class GetStartedPageService {
+  private productTypesSubject$: BehaviorSubject<SecurityProductTypes | undefined>;
+  private projectsUrlSubject$: BehaviorSubject<string | undefined>;
+  private projectFeaturesUrlSubject$: BehaviorSubject<string | undefined>;
+  private availableStepsSubject$: BehaviorSubject<StepId[]>;
+
+  public productTypes$: Observable<SecurityProductTypes | undefined>;
+  public projectsUrl$: Observable<string | undefined>;
+  public projectFeaturesUrl$: Observable<string | undefined>;
+  public availableSteps$: Observable<StepId[]>;
+
+  constructor() {
+    this.productTypesSubject$ = new BehaviorSubject<SecurityProductTypes | undefined>(undefined);
+    this.projectsUrlSubject$ = new BehaviorSubject<string | undefined>(undefined);
+    this.projectFeaturesUrlSubject$ = new BehaviorSubject<string | undefined>(undefined);
+    this.availableStepsSubject$ = new BehaviorSubject<StepId[]>([]);
+
+    this.productTypes$ = this.productTypesSubject$.asObservable();
+    this.projectsUrl$ = this.projectsUrlSubject$.asObservable();
+    this.projectFeaturesUrl$ = this.projectFeaturesUrlSubject$.asObservable();
+    this.availableSteps$ = this.availableStepsSubject$.asObservable();
+  }
+
+  getSettings() {
+    return {
+      productTypes: this.productTypesSubject$.getValue(),
+      projectsUrl: this.projectsUrlSubject$.getValue(),
+      projectFeaturesUrl: this.projectFeaturesUrlSubject$.getValue(),
+      availableSteps: this.availableStepsSubject$.getValue(),
+    };
+  }
+
+  getProductTypes() {
+    return this.productTypesSubject$.getValue();
+  }
```

Comment:
There's no point using observables if we read the values with `getValue`, that's static, if the value changes the components won't notice. We should read the observable and consume them with `useObservable` in the components.

It probably makes more sense to have separate helper hooks (`useProductTypes`, `useProjectUrl`, ...) to be consumed by the components, rather than keeping everything in a context, the context will re-render everything if any value changes.